src/lightning/pytorch/utilities/model_helpers.py

```diff
@@ -58,6 +58,24 @@ def get_torchvision_model(model_name: str, **kwargs: Any) -> nn.Module:
     return getattr(models, model_name)(**kwargs)
 
 
+class _ModuleMode:
+    """Captures the ``nn.Module.training`` (bool) mode of every submodule, and allows it to be restored later on."""
+
+    def __init__(self) -> None:
+        self.mode: Dict[str, bool] = {}
+
+    def capture(self, module: nn.Module) -> None:
+        for name, mod in module.named_modules():
```

Comment:
Should the dict be cleared before capturing in case the model changes?

Reply:
Good idea! Done and added test.